e2e/nomostest/portforwarder/port_forwarder.go

```diff
@@ -254,3 +259,37 @@ func (pf *PortForwarder) start() error {
 	}()
 	return nil
 }
+
+// signalOnFailure returns a chan which will be closed if any of the provided
+// functions return. The functions should block and only return if a failure
+// has occurred and the port-forward must be restarted
+func (pf *PortForwarder) signalOnFailure(fns ...func() error) chan struct{} {
+	restartChan := make(chan struct{})
+	// signalled indicates whether a signal has been sent
+	var signalled bool
+	var mux sync.Mutex
+	for _, fn := range fns {
+		fn := fn
+		go func() {
+			err := fn()
+			mux.Lock()
```

Comment:
wont this block the 2nd function from starting until the 1st function is done?

Reply:
No, because the assumption is that the `fn` function should be blocking. The mux stops them from entering the signalling section at the same time